chapters/annotations.tex

```diff
@@ -39,14 +39,19 @@ \section{Documentation}\label{annotations-for-documentation}\label{documentation
 The \fmtannotationindex{Documentation} annotation has the following contents, where the \lstinline!info! and \lstinline!revisions! annotations are described in \cref{annotation-info-revisions}, and the \lstinline!figures! annotation is described in \cref{annotations-for-figures}:
 \begin{lstlisting}[language=modelica]
 record Documentation
-  String info = "" "Description of the class";
-  String revisions = "" "Revision history";
+  constant String info = "" "Description of the class";
+  constant String revisions = "" "Revision history";
   Figure[:] figures = {}; "Simulation result figures";
 end Documentation;
 \end{lstlisting}
 
 How the tool interprets the information in \lstinline!Documentation! is unspecified.
 
+\subsection{How to interpret annotation definitions}
+If the annotation itself is described by a \lstinline!record! \emph{class} the annotation can be seen as a modifier for a \lstinline!record! \emph{component} with the same name.
+If the annotation is declared as a parameter the corresponding modifier must be an evaluable parameter.
+If the annotation is declared as a constant the corresponding modifier must be a literal value.
+
```

Comment:
In order to work as a fix for #3450, I think that this section also needs to explain the semantics of declared default values in the record notation.  Maybe this can be explained more generally so that it doesn't only apply to record-style definitions?

Reply:
I also think it would be nice to add the description of default values because I had to explain it to someone just a couple of months ago.

Reply:
I tried to add something for that.

Reply:
From my side, this can be resolved. Leaving open, so @henrikt-ma can still add his comments.

Reply:
Obsolete conversation.  Resolving.